templates/typescript/samples/sample-skill/src/adapters/defaultAdapter.ts

```diff
@@ -87,4 +73,61 @@ export class DefaultAdapter extends BotFrameworkAdapter {
         this.use(new EventDebuggerMiddleware());
         this.use(new SetSpeakMiddleware());
     }
+
+    private async handleTurnError(turnContext: TurnContext, error: Error): Promise<void> {
+        // Log any leaked exception from the application.
+        console.error(`[onTurnError] unhandled error : ${ error }`);
+
+        await this.sendErrorMessage(turnContext, error);
+        await this.sendEndOfConversationToParent(turnContext, error);
+        await this.clearConversationStateAsync(turnContext);
+    }
+
+    private async sendErrorMessage(turnContext: TurnContext, error: Error): Promise<void> {
+        try {
+            const telemetryException: TelemetryException = {
+                exception: error
+            };
+            this.telemetryClient.trackException(telemetryException);
+
+            // Send a message to the user.
+            await turnContext.sendActivity(this.templateEngine.generateActivityForLocale('ErrorMessage'));
+
+            // Send a trace activity, which will be displayed in the Bot Framework Emulator.
+            // Note: we return the entire exception in the value property to help the developer;
+            // this should not be done in production.
+            await turnContext.sendTraceActivity('OnTurnError Trace', error.message, 'https://www.botframework.com/schemas/error', 'TurnError');
+        }
+        catch (err) {
+            console.error(`Exception caught in sendErrorMessage : ${ err }`);
+        }
+    }
+
+    private async sendEndOfConversationToParent(turnContext: TurnContext, error: Error): Promise<void> {
+        try {
+            if (TurnContextEx.isSkill(turnContext)) {
+                // Send and EndOfConversation activity to the skill caller with the error to end the conversation
+                // and let the caller decide what to do.
+                const endOfConversation = ActivityEx.createEndOfConversationActivity();
+                endOfConversation.code = 'SkillError';
+                endOfConversation.text = error.message;
+                await turnContext.sendActivity(endOfConversation);
+            }
+        }
+        catch (err) {
+            console.error(`Exception caught in sendEndOfConversationToParent : ${ err }`);
+        }
+    }
+
+    private async clearConversationStateAsync(turnContext: TurnContext): Promise<void> {
+        try {
+            // Delete the conversationState for the current conversation to prevent the
+            // bot from getting stuck in a error-loop caused by being in a bad state.
+            // ConversationState should be thought of as similar to "cookie-state" for a Web page.
+            await this.conversationState.delete(turnContext);
+        }
+        catch (err) {
+            console.error(`Exception caught in clearConversationStateAsync : ${ err }`);
```

Comment:
```suggestion
            console.error(`Exception caught on attempting to Delete ConversationState : ${ err }`);
```

Taking into account [C#](https://github.com/microsoft/botframework-solutions/blob/v1.0/samples/csharp/skill/SkillSample/Adapters/DefaultAdapter.cs#L120)

Reply:
my bad. Will correct